### ISSUE TITLE
Printing Python sys with #p fails

### ISSUE THREAD

**Issue by eregon** [oracle/truffleruby#2027]
With GraalVM 20.1.0:
```
$ bin/ruby --jvm --polyglot -e 'p Polyglot.eval("python", "import sys;sys")'                              
-e:1:in `block in pairs_from_object': Unknown identifier: make_flags_class (NameError)
	from -e:1:in `map'
	from -e:1:in `pairs_from_object'
	from -e:1:in `foreign_inspect_nonrecursive'
	from -e:1:in `block in foreign_inspect'
	from -e:1:in `detect_recursion'
	from -e:1:in `foreign_inspect'
	from -e:1:in `block in p'
	from -e:1:in `each'
	from -e:1:in `p'
	from -e:1:in `<main>'
```

I think we should always use `InteropLibrary#toDisplayString()` for `inspect`, but that means quite a few changes to our tests and logic.
For `to_s`, I'm not sure what's right, probably the existing logic for `inspect` is suitable.

/cc @fniephaus @stefreschke @jouhlmann

**Comment by eregon** [comment]
After discussing with multiple people about this, the consensus is:
* `to_s` calls `asString` if `isString`, and otherwise `toDisplayString` (new)
* `inspect` shows a Ruby-like inspect String including members, array elements, etc.

**Comment by eregon** [comment]
The specific issue with the `sys` module is solved in 8f5a0809d9afcd39a98cdb8a1432c51115f1e3a0 (the member was listed but not readable).

**Comment by eregon** [comment]
97c724fe38075a6419646b1bd00a97cf9f32b474 reworked how `to_s` and `inspect` work on foreign objects as described above.

For the case of the `sys` module from Python:
```
$ bin/ruby --jvm --polyglot -e 'value = Polyglot.eval("python", %{import sys; sys}); puts value; puts; p value;'
```
to_s:
```
#<Python <module object at 0x14e93c46>>
```

inspect:

#<Python module:0x14e93c46 copyright="Copyright (c) Oracle and/or its affiliates. Licensed under the Universal Permissive License v 1.0 as shown at http://oss.oracle.com/licenses/upl.", stdin=[...], path_importer_cache={...}, stdout=[...], prefix="/home/eregon/code/truffleruby-ws/truffleruby/mxbuild/truffleruby-jvm-py/jre/languages/python", hexversion=50856608, ps2="... ", ps1=">>> ", path=[...], builtin_module_names=[...], excepthook={...}, dont_write_bytecode=true, getrecursionlimit={...}, pycache_prefix={...}, getsizeof={...}, addaudithook={...}, getfilesystemencodeerrors={...}, _base_executable="", stderr=[...], version="3.8.2 (Sat Jun 06 18:59:57 CEST 2020)\n[Graal, Interpreted, Java 1.8.0_252]", modules={...}, _getframe={...}, exit={...}, base_prefix="/home/eregon/code/truffleruby-ws/truffleruby/mxbuild/truffleruby-jvm-py/jre/languages/python", maxsize=2147483647, path_hooks=[...], warnoptions=[...], exec_prefix="/home/eregon/code/truffleruby-ws/truffleruby/mxbuild/truffleruby-jvm-py/jre/languages/python", flags=[...], argv=[...], getfilesystemencoding={...}, platform="linux", hash_info=[...], displayhook={...}, audit={...}, byteorder="little", int_info=[...], maxunicode=1114111, meta_path=[...], exc_info={...}, intern={...}, unraisablehook={...}, implementation={...}, base_exec_prefix="/home/eregon/code/truffleruby-ws/truffleruby/mxbuild/truffleruby-jvm-py/jre/languages/python", version_info=[...], executable="", abiflags="", float_info=[...], breakpointhook={...}, getdefaultencoding={...}>
